### ISSUE TITLE
False `UndefinedMagicMethod` on builder calls when a model declares `@mixin \…\Model` / `@mixin \Eloquent`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1215]
## Description

A model that declares a hand-written `@mixin` docblock pointing at a class **without** query-builder
methods — `@mixin \Illuminate\Database\Eloquent\Model` or the laravel-ide-helper alias `@mixin \Eloquent`
— loses the plugin's builder-method resolution. Every builder call (`find`, `where`, `get`, `first`,
`pluck`, `create`, `whereHas`, `firstOrNew`, ...) then reports `UndefinedMagicMethod`.

This convention is widespread: laravel-ide-helper writes `@mixin \Eloquent` on every model, and the
October/Winter/TastyIgniter lineage hand-writes `@mixin \Illuminate\Database\Eloquent\Model`. Affected
projects see false positives across their entire model layer, with `mixed` cascading into downstream
`PossiblyNull*` / `InvalidArgument`.

Found auditing **tastyigniter/core** (Laravel 12): **148 `UndefinedMagicMethod`** — the single largest
false-positive bucket — all traced to this one docblock pattern.

## Reproduction

```php
use Illuminate\Database\Eloquent\Model;

/** @mixin \Illuminate\Database\Eloquent\Model */
class Status extends Model { protected $table = 'statuses'; }

Status::find(1);            // UndefinedMagicMethod: Status::find does not exist
Status::where('id', 1);    // UndefinedMagicMethod: Status::where does not exist
```

Behavior matrix (plain `extends Illuminate\Database\Eloquent\Model`, errorLevel 3):

| Model class docblock                           | `::find()` / `::where()`      |
|------------------------------------------------|-------------------------------|
| _(none)_                                        | resolves                      |
| `@mixin \Illuminate\Database\Eloquent\Builder`  | resolves                      |
| `@mixin \Illuminate\Database\Eloquent\Model`    | `UndefinedMagicMethod`        |
| `@mixin \Eloquent`                              | `UndefinedMagicMethod`        |

A custom base model / custom Eloquent builder are **not** required to reproduce; isolated they behave
correctly. The trigger is solely the model's own `@mixin` docblock.

## Expected Behavior

Builder methods resolve on the model regardless of a user-authored `@mixin`. `@mixin \...\Model` on a
class that already *is* a `Model` is a semantic no-op self-mixin and should stay harmless.

## Actual Behavior

The user `@mixin` **replaces** the plugin's injected builder `@mixin` (Psalm `@mixin` does not merge —
same no-merge behavior handled for builder subclasses in #1140 / PR #1145). When it points at a class
lacking query methods, `find`/`where`/etc. resolve to nothing → `UndefinedMagicMethod`.

## Proposed Fix

Append the plugin's builder `@mixin` rather than let a user `@mixin` replace it — mirror
`BuilderSubclassQueryMixinHandler` (#1145), but for **models** that carry their own `@mixin`.
Alternatively, a model-builder method-existence provider that a user `@mixin` cannot defeat.

The `@mixin` no-merge is Psalm core behavior; this case is plugin-fixable without an upstream change
(the sibling builder-subclass case already is).

## Regression test

A drop-in `tests/Type/tests/MixinModelBuilderTest.phpt` (currently failing, `--EXPECT--` empty for the
desired post-fix state) is available.

**Comment by alies-dev** [comment]
Folding in #1171 (ide-helper `@method static \Builder|Model withTrashed()`), which overlaps this issue. Two sub-cases there:

**Sub-case A — static chain (`@mixin \Eloquent` required).** `Model::withTrashed()->where(...)` → `UndefinedMagicMethod - Model::where`. Drop the `@mixin \Eloquent` line and it disappears. Same root cause as this issue: the user `@mixin` replaces the plugin's injected builder `@mixin` (Psalm no-merge). The fix proposed here (append the builder `@mixin` for models that carry their own) should resolve this too. Regression test should include a SoftDeletes + `@mixin \Eloquent` model.

**Sub-case B — instance call, mixin-INDEPENDENT (the distinct residual).** `Model::query()->withTrashed()` → `mixed` instead of `Builder<Model>`. Reproduces with a plain `Model` and a single ide-helper annotation, **no mixin**:

```php
/** @method static \Illuminate\Database\Eloquent\Builder|ReproUser withTrashed() */
class ReproUser extends Model { use SoftDeletes; }

ReproUser::query()->withTrashed();  // mixed; expected Builder<ReproUser>
```

Root cause is different from the `@mixin` no-merge: the non-generic `\Builder|Model` union in the model's own `@method static` pseudo-method defeats builder-instance trait-macro resolution and collapses to `mixed`, which is strictly worse than the macro-signature fallback (`Builder<Model>`). This will NOT be fixed by appending the builder `@mixin` — it needs the trait-macro resolver to prefer its own `Builder<Model>` inference over a non-generic `\Builder|Model` `@method static` union (or ignore the union arm). Track as a second checkbox under this issue.

Distinct from #1054 (generic scope-prefixed `@method static Builder<static> scopeXxx()`) and #635 (SoftDeletes on base Builder with NO annotation). Trigger for sub-case B is the presence of default ide-helper `@method static` output.

Closing #1171 as covered here.

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.
